improver_tests/psychrometric_calculations/cloud_top_temperature/test_CloudTopTemperature.py

```diff
@@ -0,0 +1,158 @@
+    temperatures = np.array([300, 286, 280, 274, 267, 262, 257, 245], dtype=np.float32)
+    data = np.broadcast_to(
+        temperatures.reshape((1, len(temperatures), 1, 1)), (2, len(temperatures), 2, 2)
+    )
+    t_cube = set_up_variable_cube(
+        data + profile_shift,
+        pressure=True,
+        height_levels=np.arange(100000, 29999, -10000),
+        name="temperature_on_pressure_levels",
+        units="K",
+        attributes=LOCAL_MANDATORY_ATTRIBUTES,
+    )
+    return t_cube
+
+
+def metadata_ok(cct: Cube, baseline: Cube, model_id_attr=None) -> None:
+    """
+    Checks convective cloud top temperature Cube long_name, units and dtype are as expected.
+    Compares with baseline to make sure everything else matches.
+
+    Args:
+        cct: Result of ConvectiveCloudTop plugin
+        baseline: A cube with the expected coordinates and attributes.
+
+    Raises:
+        AssertionError: If anything doesn't match
+    """
+    assert cct.long_name == "air_temperature_at_convective_cloud_top"
+    assert cct.units == "K"
+    assert cct.dtype == np.float32
+    for coord in cct.coords():
+        base_coord = baseline.coord(coord.name())
+        assert cct.coord_dims(coord) == baseline.coord_dims(base_coord)
+        assert coord == base_coord
+    for attr in MANDATORY_ATTRIBUTES:
+        assert cct.attributes[attr] == baseline.attributes[attr]
+    all_attr_keys = list(cct.attributes.keys())
+    if model_id_attr:
+        assert cct.attributes[model_id_attr] == baseline.attributes[model_id_attr]
+        mandatory_attr_keys = [k for k in all_attr_keys if k != model_id_attr]
+    else:
+        mandatory_attr_keys = all_attr_keys
+    assert sorted(mandatory_attr_keys) == sorted(MANDATORY_ATTRIBUTES)
+
+
+@pytest.mark.parametrize(
+    "ccl_t, ccl_p, profile_shift, expected",
+    (
+        (290, 95000, 0, 264.575),
+        (288.12, 90000, 0, 264.575),
+        (290, 95000, -4, 254.698),
+        (288.17, 90000, -4, 254.698),
+        (286, 91000, 0, False),
+    ),
+)
+def test_basic(ccl, temperature, ccl_t, ccl_p, expected):
+    """
+    When the profile is shifted to be colder, the same saturated ascents
+    reach a higher, and therefore colder, level.
+    The different CCL values show that starting at a higher point on the same
+    profile has no impact on the final result.
+    The last test is for a case where convection does not occur and the result is masked.
+    """
+    ccl.data = np.full_like(ccl.data, ccl_t)
+    ccl.coord("air_pressure").points = np.full_like(
+        ccl.coord("air_pressure").points, fill_value=ccl_p
+    )
+    result = CloudTopTemperature()(ccl, temperature)
+    metadata_ok(result, ccl)
+    if expected:
+        assert not result.data.mask.any()
+        np.testing.assert_allclose(result.data, expected, atol=1e-2)
+    else:
+        assert result.data.mask.all()
+
+
+@pytest.mark.parametrize("profile_shift", (0,))
```

Comment:
It is! `t_cube_fixture`, which is invoked here as `temperature`, requires a `profile_shift`, because I haven't found a way to define fixtures with default values.

Reply:
Oh yes I see. That seems odd as  its not called directly into the function its actually related to but I see how it's being used.